src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -1271,6 +1274,35 @@ private void cleanupChatMessages()
         }
     }
 
+    @Override
+    public void setCustomNameTag(final String name)
+    {
+        if(citizenData != null && name != null && Configurations.gameplay.allowGlobalNameChanges >= 0)
+        {
+            if(Configurations.gameplay.allowGlobalNameChanges == 0 &&
+                    Arrays.stream(Configurations.gameplay.specialPermGroup).anyMatch(owner -> owner.equals(colony.getPermissions().getOwnerName())))
+            {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00108)